Tests/OpenCombineTests/PublisherTests/FutureTests.swift

```diff
@@ -0,0 +1,129 @@
+//
+//  FutureTests.swift
+//  
+//
+//  Created by Max Desiatov on 24/11/2019.
+//
+
+import XCTest
+
+#if OPENCOMBINE_COMPATIBILITY_TEST
+import Combine
+#else
+import OpenCombine
+#endif
+
+@available(macOS 10.15, iOS 13.0, *)
+final class FutureTests: XCTestCase {
+    private typealias SUT = Future<Int, TestingError>
+
+    func testFutureSuccess() {
+        var promise: SUT.Promise?
+
+        let future = SUT { promise = $0 }
+
+        let subscriber = TrackingSubscriber(receiveSubscription: { subscription in
+            subscription.request(.unlimited)
+        })
+
+        future.subscribe(subscriber)
+        promise?(.success(42))
+
+        XCTAssertEqual(subscriber.history, [
+            .subscription("Future"),
+            .value(42),
+            .completion(.finished)
+        ])
+    }
+
+    func testFutureFailure() {
+        var promise: SUT.Promise?
+
+        let future = SUT { promise = $0 }
+
+        let subscriber = TrackingSubscriber(
+            receiveSubscription: { subscription in
+                subscription.request(.unlimited)
+            }, receiveValue: { _ in
+                XCTFail("no value should be returned")
+                return .unlimited
```

Comment:
The conventional thing to do is to return `.none` to indicate that we don't want any more values than we already requested.
```suggestion
                return .none
```